pkg/process/namespace.go

```diff
@@ -0,0 +1,42 @@
+package process
+
+import (
+	"github.com/grafana/tanka/pkg/kubernetes/manifest"
+)
+
+const (
+	// AnnotationNamespaced can be set on any resource to override the decision
+	// whether 'metadata.namespace' is set by Tanka
+	AnnotationNamespaced = MetadataPrefix + "/namespaced"
+)
+
+// Namespace injects the default namespace of the environment into each
+// resources, that does not already define one. AnnotationNamespaced can be used
+// to disable this per resource
+func Namespace(list manifest.List, def string) manifest.List {
```

Comment:
Perhaps this method could be named a little more descriptively. Maybe `AddNamespace` or `EnsureNamespace`.

Reply:
I tried to name the methods inside of process by single verbs, as they are a chain of actions:

- Extract
- Unwrap
- Namespace
- Label
- Filter
- Sort

(Hoping "to namespace something" is actually a verb)